### PR TITLE
Split platform

### LINKED ISSUE #766: 4.3 Split Platforms
## 🐞 Overview
Backend processing to split one platform into several instances.

The underlying issue here relates to the "Merlin" helicopter:
image

When data from this type of helicopter has been imported, it has mistakenly been imported into a single "Merlin" platform. The correct use of Pepys would have been to have one "Merlin" platform per aircraft, each uniquely identified by the airframe number (ZH860 in the following image).
image

So, this task is to allow analysts to split a single Platform into multiple ones.  It is assumed that one Merlin datafile will only contain data for a single Merlin.

## API

* Receives: PlatformID
* Processing:
    - find list of datafiles for this platform
    - duplicate platform, once per datafile, using date-time of first measurement for datafile as "identifier" yymmdd-hhmmss
    - re-assign datafile to the new platform
    - duplicate sensor(s) in new platform as necessary to ensure sensor is present for measurements in that datafile. 
* delete original platform
* log changes (splitting original platform, creating new platform)

## 🔗 Feature

#600 Maintenance tools

## 🔢 Acceptance criteria

* Database initialised to contain several datafiles for same platform
* call split function, passing platform id
* verify there's now one platform per datafile, and that each platform has the correct sensor

### PR DESCRIPTION
**Pull request by BarisSari**
## 🧰 Issue
Fixes #766 

## 🔨Work carried out:
- [x] Update `resolve_platform` method (it fails when there are more than 1 Platform with the same name, nationality, and platform_type)
- [x] Fix new tests
- [x] Remove duplicated code in `split_platform`
- [x] Increase "identifier" field from 10 chars to 30 chars
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR, adding `interactive_review` if reviewers will need to see UI
- [x] I have extended/updated the documentation in `\docs` folder
- [x] Any database changes are recorded in the Log/Changes tables
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

### CODE REVIEW

**Review by robintw — CHANGES_REQUESTED** [review]
This looks good in general @BarisSari - but I'm wondering whether it properly deals with Comment entries? They have a direct platform_id reference, rather than referring to a Sensor object - would they have their platform_id entry updated correctly? Could you add a test with a comment entry?

**Comment by BarisSari** [comment]
> This looks good in general @BarisSari - but I'm wondering whether it properly deals with Comment entries? They have a direct platform_id reference, rather than referring to a Sensor object - would they have their platform_id entry updated correctly? Could you add a test with a comment entry?

Sure, @robintw. I'll add a test. If it fails, I'll update the code.

File: pepys_import/database/default_data/Help Texts.csv [hunk 1]
```diff
@@ -18,6 +18,6 @@ FUZZY_SEARCH_SENSOR_TYPE,"Specify the type of sensor."
 FUZZY_SEARCH_NATIONALITY,"Specifythe nationality of this platform"
 KEEP_PLATFORM_AS_SYNONYM,"Add platform name as a synonym. Warning: This should only be done when platform name is a completely unique identifier for this platform not a name that could be shared across platforms of different nationalities."
 DID_NOT_SELECT_EXISTING,"You did not select an existing value. Please search again."
-PLATFORM_IDENTIFIER,"The identifier of platform, such as a Pennant number for a warship or tail number in an aircraft. Where an identifer isn't available for that platform-type, the platform name should be entered as identifier. Up to 10 characters are allowed."
+PLATFORM_IDENTIFIER,"The identifier of platform, such as a Pennant number for a warship or tail number in an aircraft. Where an identifer isn't available for that platform-type, the platform name should be entered as identifier. Up to 30 characters are allowed."
```
Comment (IanMayo) [inline pepys_import/database/default_data/Help Texts.csv:21]: well spotted @BarisSari  :-)

Comment (BarisSari) [reply pepys_import/database/default_data/Help Texts.csv:21]: Thanks. I checked the "10" string in the whole project :)

**Review by robintw — APPROVED** [review]
Looks good, thanks

**Comment by IanMayo** [comment]
@BarisSari - Codacy moaned about the multiline doc string at the first line of the Alembic files:
image

The error does appear to refer to an existing standard:
image

Is the whole file auto-generated?  If that's the case then we have no control over the comment format - and I can configure Codacy to ignore that issue. But, if you've written the starting comment, how would you feel about updating to match the standard?

**Comment by BarisSari** [comment]
It was generated by Alembic(https://github.com/debrief/pepys-import/pull/770/commits/8d8129eacd6f612bb346e67a062afa76834e35d0). Then, I changed it to solve D213 issue. But now, we have another issue: D212 😅

**Comment by IanMayo** [comment]
> It was generated by Alembic([8d8129e](https://github.com/debrief/pepys-import/commit/8d8129eacd6f612bb346e67a062afa76834e35d0)). Then, I changed it to solve D213 issue. But now, we have another issue: D212 😅 Maybe we can ignore D213 and D212.

Aah, they're mutually exclusive: https://stackoverflow.com/questions/45990301/pep257-d212-and-d213-conflicts

**Review by IanMayo — APPROVED** [review]
Great. Have relaxed Codacy issue.

**Pull request merged**